R/sql_methods.R

```diff
@@ -50,13 +50,23 @@ read_sql <- function(sql, dbname = NULL, schemaname = NULL){
  # Unlisting list column types. Also substituting NULL values to NA to remain consistent.
  for (i in seq(1,ncol(result))){
    if (typeof(result[,i]) == "list"){
-     if (class(result[,i][[1]])[1] == "datetime.date"){
-       
-       # returning dates in posixct format if stored as dates.
-       result[,i] <- as.Date(unlist(lapply(result[,i], function(x) {if (is.null(x)){NA} else {as.character(x)}})))
+     
+     # finding the first not null element in the list to determine its class. If it is python's datetime.date, returning it as Date.
+     k = 1
+     
+     while (is.null(result[,i][[k]])) {
```

Comment:
it would be much nicer to have this as a single while is.null(result[,i][[k]]) && k < length(result[,i])

Reply:
Corrected.